### PR TITLE
Signal Messenger: add text_mode and improvements

### PR DESCRIPTION
**Pull request by r-xyz**
## Proposed change

Adding documentation of new [`text_mode` option](https://github.com/home-assistant/core/pull/117148) and improves overall Signal Messenger docs.

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/117148
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **Documentation**
  - Enhanced `signal_messenger` integration guide with new setup requirements and reorganized content.
  - Updated configuration options, including sender/recipient formats and examples for sending notifications.
  - Added instructions for triggering events based on Signal message reception.
  - Improved clarity and organization for better user guidance.

### CODE REVIEW

File: source/_integrations/signal_messenger.markdown [hunk 1]
```diff
@@ -20,6 +20,7 @@ The `signal_messenger` {% term integration %} uses the [Signal Messenger REST AP
 
 ## Setup
  
+### Requirements
```
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: Implemented.

File: source/_integrations/signal_messenger.markdown [hunk 2]
```diff
@@ -149,4 +129,51 @@ action:
...
+
+**Notes:**
+
+- To attach files from outside of Home Assistant, the URLs must be reachable and added to the [`allowlist_external_urls`](/integrations/homeassistant/#allowlist_external_urls) list.
+
+- There is a 50MB size limit for attachments retrieved via URLs. 
+
+## Triggering events based on Signal message reception
+
+You can use Signal Messenger REST API as a Home Assistant trigger. In this example, we will make a simple chatbot. If you write anything to your Signal account linked to Signal Messenger REST API, the automation gets triggered, with the condition that the number (attribute source) is correct, to take action by sending a Signal notification back with a "Message received!".
```
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: fixed.

File: source/_integrations/signal_messenger.markdown [hunk 3]
```diff
@@ -19,17 +19,17 @@ related:
...
+The requirements are:
 
-Please follow those [instructions](https://github.com/bbernhard/signal-cli-rest-api/blob/master/doc/HOMEASSISTANT.md), to set up the Signal Messenger REST API. 
+- You need to set up the Signal Messenger REST API.
+- You need a spare phone number to register with the Signal Messenger service.
 
+Please follow those [instructions](https://github.com/bbernhard/signal-cli-rest-api/blob/master/doc/HOMEASSISTANT.md), to set up the Signal Messenger REST API.
 
-## Configuration
+### Configuration Variables
```
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: Suggestion implemented in c5bc61fef43fcd32df98a140affa279f06d3bb3d

File: source/_integrations/signal_messenger.markdown [hunk 4]
```diff
@@ -60,67 +56,49 @@ url:
   required: true
   type: string
 number:
-  description: The sender number.
+  description: The sender number, [see Sender/recipient formats](#senderrecipient-formats).
```
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: fixed in c5bc61fef43fcd32df98a140affa279f06d3bb3d

File: source/_integrations/signal_messenger.markdown [hunk 5]
```diff
@@ -60,67 +56,49 @@ url:
...
-    data:
-      message: "Message received!"
-```
+## Notification Service
 
-## Examples
+### Examples
 
-A few examples on how to use this integration as actions in automations.
+A few examples on how to use this integration to send notifications from automations..
```
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: fixed in c5bc61fef43fcd32df98a140affa279f06d3bb3d

File: source/_integrations/signal_messenger.markdown [hunk 6]
```diff
@@ -149,4 +129,51 @@ action:
...
+| ----------- | -------- | ------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `urls` | **required** | -  | List of URLs of files to be attached. |
+| `verify_ssl` | *optional* | true  | Accepted values are `true`, `false`. You can set it to `false` to ignore SSL errors. |
+| `text_mode` | *optional* | normal | Accepted values are `normal` or `styled`. If set to `styled`, additional text formatting is enabled (*`*italic*`*, **`**bold**`**, and ~~`~strikethrough~`~~). |
+
+**Notes:**
+
+- To attach files from outside of Home Assistant, the URLs must be reachable and added to the [`allowlist_external_urls`](/integrations/homeassistant/#allowlist_external_urls) list.
+
+- There is a 50MB size limit for attachments retrieved via URLs.
```
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: fixed

File: source/_integrations/signal_messenger.markdown (hunk 2 above)
[the remark this answers is not in the document]
Comment (r-xyz) [reply source/_integrations/signal_messenger.markdown]: Thanks.
All changes have been implemented and checks passed. Could you please approve the PR or ask an human maintainer to review it? 
Thank you.

**Comment by r-xyz** [comment]
@coderabbitai review

**Review by frenck — CHANGES_REQUESTED** [review]
This PR is changing our regular structure.

Please remove any other improvements from this PR and keep it to just the change that is happening in the parent PR.

If you want to do additional rewriting of this page, that should be done in a different PR (against the `current` branch).

../Frenck

**Review by frenck — APPROVED** [review]
Thanks, @r-xyz & @c0ffeeca7 👍 

../Frenck

**Pull request merged**